### PR TITLE
object detection examples notebook

### PR DESCRIPTION
**Pull request by aditya1503**

### CODE REVIEW

**Review by ulya-tkch — APPROVED** [review]
Approved after addressing PR comments on wording. Great job!

**Comment by jwmueller** [comment]
reminder for myself: add to table of contents right before merge

File: object_detection/detectron2_training.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training.ipynb:2]: need requirements.txt file. Make sure the first line of requirements file says:

\# This notebook currently requires the latest developer version of cleanlab: pip install git+https://github.com/cleanlab/cleanlab.git

Comment (jwmueller) [reply object_detection/detectron2_training.ipynb:2]: need a README inside the object_detection folder that in 1-line describes what is difference between the two notebooks

File: object_detection/detectron2_training-kfold.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training-kfold.ipynb]:
```suggestion
    "# Fitting a (Detectron2) Object Detection Model via K-Fold Cross-Validation to produce Out-of-sample Predictions for every Image in a Dataset \n",
```

File: object_detection/detectron2_training.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training.ipynb]:
```suggestion
    "This notebook demonstrates how to train a [Detectron2](https://github.com/facebookresearch/detectron2/) model on object detection datasets and produce predictions required to run cleanlab's tutorial on detecting label errors in object detection data.  Note that this notebook fits the model to an entire training set and produces predictions on a held-out validation set. Thus these predictions are only *out-of-sample* for the validation data, and should ideally *only* be used to find mislabeled images amongst the validation set. To instead find mislabeled images amongst an entire dataset, see the analogous notebook in this folder which uses K-fold cross-validation to produce out-of-sample predictions for every image in the dataset.\n",
    "\n",
    "In object detection data, each image is annotated with multiple bounding boxes.  Each bounding box surrounds a physical object within an image scene, and is annotated with a given class label. Using this labeled data, we train a model to predict the locations and classes of objects in an image. The trained model can subsequently be used to identify mislabeled images, which when corrected, allow you to train an even better model without changing your training code! \n",
```

File: object_detection/detectron2_training.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training.ipynb]:
```suggestion
    "Here we will fit a state-of-the-art neural network trained starting from a pretrained [X-101](https://github.com/facebookresearch/detectron2/blob/main/MODEL_ZOO.md#imagenet-pretrained-models) network backbone. First let's import the required packages and download the data."
```

File: object_detection/detectron2_training.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training.ipynb]:
```suggestion
    "Here the number of worker threads is set to 2 and the batch size is set to 2. The learning rate and maximum number of iterations are also specified. You'll want to tinker with these values to get the best performance for your own data.\n",
    "The model is initialized from the COCO-Detection model zoo and the output directory for the trained model is created. Finally, this configuration is passed to the DefaultTrainer class for training the object detection model.\n",
```

File: object_detection/detectron2_training.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training.ipynb]:
```suggestion
    "cfg.SOLVER.BASE_LR = 0.00025  # IMPORTANT: pick a good Learning Rate for your dataset \n",
```

File: object_detection/detectron2_training.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training.ipynb]:
```suggestion
    "The code below defines a function \"format_detectron2_predictions\" to convert the prediction output of Detectron2 to a format that can be used by Cleanlab for identifying label errors. This function accepts the predicted object instances and the number of classes as inputs. It processes the predicted bounding boxes and prediction-confidence for each instance, and outputs a list of numpy arrays containing the bounding boxes and prediction-confidence for each class."
```

Comment (jwmueller) [reply object_detection/detectron2_training.ipynb]: is it for each class? I thought it was just for the predicted (i.e. most likely) class

File: object_detection/detectron2_training.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training.ipynb]:
```suggestion
    "To perform inference and testing on the tutorial notebook linked here, we utilize a limited portion of the validation set of COCO 2017. To find label errors in this subset, please run our [tutorial](https://docs.cleanlab.ai/stable/index.html) notebook on [Finding Label Errors in Object Detection Datasets](https://github.com/cleanlab/cleanlab/blob/master/docs/source/tutorials/object_detection.ipynb)."
```

File: object_detection/detectron2_training-kfold.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training-kfold.ipynb]:
```suggestion
    "This notebook demonstrates how to fit a [Detectron2](https://github.com/facebookresearch/detectron2/) model on object detection datasets via *cross-validation*, and produce *out-of-sample* predictions required to run cleanlab for detecting label errors in such data.  Out-of-sample predictions (from a copy of the model that was never trained on the image it is producing a prediction for) are less subject to overfitting and can more accurately detect label issues, but are harder to obtain for every image in an entire dataset unless we employ K-fold cross-validation as demonstrated here.\n",
    "\n",
    "Thus run this notebook if you wish to find every mislabeled image an a dataset. Once you have the out-of-sample predictions produced from this notebook, you can directly run the cleanlab [tutorial](https://docs.cleanlab.ai/stable/index.html) notebook on [Finding Label Errors in Object Detection Datasets](https://github.com/cleanlab/cleanlab/blob/master/docs/source/tutorials/object_detection.ipynb). Even though the cleanlab tutorial is focused on finding issues in only the validation subset of COCO, you can easily update the tutorial to run on the full dataset for which we obtain predictions via cross-validation in this notebook.\n",
    "\n",
    "In object detection data, each image is annotated with multiple bounding boxes.  Each bounding box surrounds a physical object within an image scene, and is annotated with a given class label. Using this labeled data, we train a model to predict the locations and classes of objects in an image. The trained model can subsequently be used to identify mislabeled images, which when corrected, allow you to train an even better model without changing your training code! \n",
    "\n",
    "Here we fit a state-of-the-art neural network trained starting from a pretrained [X-101](https://github.com/facebookresearch/detectron2/blob/main/MODEL_ZOO.md#imagenet-pretrained-models) network backbone. First let's import the required packages and download the data."
```

File: object_detection/detectron2_training-kfold.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training-kfold.ipynb]:
```suggestion
    "* Finally we save the training and test data as separate JSON files and print information about each fold."
```

File: object_detection/detectron2_training-kfold.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training-kfold.ipynb]:
```suggestion
    "NUM_FOLDS = 5  # Reduce this to get faster results but potentially less accurate predictions \n",
```

File: object_detection/detectron2_training-kfold.ipynb
Comment (jwmueller) [inline object_detection/detectron2_training-kfold.ipynb:1]: where appropriate also incorporate my suggested changes for the other notebook into this one

File: object_detection/detectron2_training.ipynb
Comment (aditya1503) [reply object_detection/detectron2_training.ipynb]: yes most confident class

Comment (jwmueller) [reply object_detection/detectron2_training.ipynb]: then please delete: "for each class" at the end

File: object_detection/requirements.txt
Comment (jwmueller) [inline object_detection/requirements.txt]:
```suggestion
# Note: for now, these examples only work with the developer version of cleanlab: `pip install git+https://github.com/cleanlab/cleanlab.git`
```

**Review by jwmueller — APPROVED** [review]
LGTM

**Pull request merged**
